notifications_utils/letter_timings.py

```diff
@@ -68,3 +68,36 @@ def get_next_royal_mail_working_day(date):
         earliest_delivery=set_gmt_hour(earliest_delivery, hour=16),
         latest_delivery=set_gmt_hour(latest_delivery, hour=16),
     )
+
+
+def letter_can_be_cancelled(notification_status, notification_created_at):
+    '''
+    If letter does not have status of created or pending-virus-check
+        => can't be cancelled (it has already been processed)
+
+    If it's after 5.30pm local time and the notification was created today before 5.30pm local time
+        => can't be cancelled (it will already be zipped up to be sent)
+    '''
+    if notification_status not in ('created', 'pending-virus-check'):
+        return False
+
+    if _is_after_half_five_local_time() and _notification_created_before_half_five_today(notification_created_at):
+        return False
+
+    return True
+
+
+def _is_after_half_five_local_time():
+    current_utc_datetime = datetime.utcnow()
+    bst_time = convert_utc_to_bst(current_utc_datetime).time()
+
+    return bst_time >= time(17, 30, 0, 0)
```

Comment:
API has a config variable for this called `LETTER_PROCESSING_DEADLINE`. If this is going to be used across admin and the API maybe we should define it as a constant in this module and use that in the API instead.

Reply:
👍 Now updated